src/sortStudents.ts

```diff
@@ -1,7 +1,71 @@
 // describe Student type
 // create and export SortType enum
 // create SortOrder type
+interface Student {
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
+}
+
+export enum SortType {
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
+}
+
+type SortOrder = 'asc' | 'desc';
+
+function getAverage(student: Student): number {
+  /* eslint-disable max-len */
+  const average: number = student.grades.reduce((total: number, grade: number) => (
+    total + grade)) / student.grades.length;
+
+  return average;
+}
+
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  const studentsCopy: Student[] = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      return order === 'asc'
+        ? studentsCopy.sort((first: Student, second: Student) => (
+          first.name.localeCompare(second.name)))
```

Comment:
use sortBy everywhere instead object key names

```suggestion
          first[sortBy]localeCompare(second[sortBy])))
```